docs/oci-storage.md

```diff
@@ -1,40 +1,149 @@
+:::note
+Fleet does not fall back to etcd if the secret is missing or invalid. Instead, it logs an error and skips the deployment.
+:::
 
-Once the OCI registry is enabled, Fleet will use it as the source for storing `Bundle` resources.
-When Fleet can't access the OCI registry, it won't fall back to default `etcd` storage. Instead, it will log errors so they can be fixed.
+## **Prerequisites**
 
-## Configuring the OCI registry
+* A running OCI registry.  
+* A Kubernetes secret with valid credentials.  
+* A Fleet installation (v2.12.0 or later) .
 
-OCI registry values should be configured as an extra section in the `GitRepo` yaml.
+## Creating an OCI Storage Secret
 
-There are the fields involved:
+OCI Storage is automatically enabled in Fleet when a valid secret is present in the same namespace as the GitRepo. To enable OCI Storage, you have to define a specific kind of secret. There are two ways of defining secrets
+
+* **Global secret:** A secret exactly named `ocistorage` in the same namespace as your GitRepos.
+  * This is the fallback secret. If no GitRepo-level secret is specified, Fleet uses this secret for all GitRepos in the namespace.  
+* **GitRepo-level secret:** A custom secret for specific GitRepo resouces.
+  * This is a user-defined secret can have any name and must be referenced in the `GitRepo` resource. 
+
+### GitRepo Configuration Example
+
+If a secret named `ocistorage` exists in a namespace. Fleet automatically uses it as the OCI registry for all GitRepo resources in that namespace, and you don’t need to add `ociRegistrySecret` in every GitRepo.
+
+You can explicitly specify a different secret in the GitRepo. The GitRepo spec is extended with information about the OCI Registry. You have to update your GitRepo resource to include a reference to the secret:
+
+```yaml
+metadata:
+  name: coffee-shop-deployments
+  namespace: fleet-local
+spec:
+  repo: https://github.com/example/coffee-shop-fleet
+  branch: main
+  paths:
+    - ./manifests
+  ociRegistrySecret: ocistorage
+```
+
+The `ociRegistrySecret` references a secret in the same namespace, which contains the address and credentials needed to read/write to the OCI registry. 
 
 ```yaml
-// when ociRegistry is defined Fleet will use oci registry as storage
-ociRegistry:
-    // reference is the OCI registry url.
-    reference: "docker.io/your-user-here"
-    // secret name where the credentials for the OCI registry are.
-    // expects a generic secret with username and password keys set.
-    authSecretName: oci-secret
-    // basicHTTP allows Fleet to use basic http connections to communicate
-    // with the registry (defaults to false)
-    basicHTTP: false
-    // insecureSkipTLS allows connections to the OCI registry
-    // without certs (defaults to false)
-    insecureSkipTLS: false
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ocistorage
+  namespace: fleet-local
+type: fleet.cattle.io/bundle-oci-storage/v1alpha1
```

Comment:
We'd need to be more explicit saying that the type in **enforced** to be `fleet.cattle.io/bundle-oci-storage/v1alpha1` otherwise the secret will be rejected by Fleet